src/kiwixmessagebox.cpp

```diff
@@ -45,3 +55,11 @@ void showInfoBox(QString title, QString text, QWidget *parent)
         dialog->deleteLater();
     });
 }
+
+KiwixMessageBox::Result showKiwixMessageBox(QString title, QString text, QWidget *parent, QString leftTitle, QString rightTitle)
+{
+    KiwixMessageBox *dialog = new KiwixMessageBox(title, text, false, parent, leftTitle, rightTitle);
+    dialog->setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
This is an interesting finding. Can't we use the same trick in `showInfoBox()`? Or maybe even move it into the constructor of `KiwixMessageBox`?

Reply:
Moving into the constructor is a nice idea, I did it in the new fixup. 
`showConfirmBox()` also uses execDialog() now - It looks more natural to me that a confirmBox() is high prio and should be handled before interacting with other windows (hence the `exec()` call)

Reply:
Now that I looked into the documentation of `QDialog::exec()` and `Qt::WA_DeleteOnClose`, I have some doubts. When `QDialog::exec()` returns the dialog should have closed and hence deleted. Therefore   `KiwixMessageBox::execDialog()` will execute its last few instructions on a deleted object. Am I wrong?

Reply:
Also, I am also concerned with `QDialog::exec()` being a blocking function. I wonder how download updater code will react to the the main thread being blocked.

Reply:
> Now that I looked into the documentation of QDialog::exec() and Qt::WA_DeleteOnClose ..........

Sorry for the delay, life happened!
You're right, I looked into qdialog's source code and [when deleteOnClose is enabled](https://codebrowser.dev/qt5/qtbase/src/widgets/dialogs/qdialog.cpp.html#612) it deletes it instantly. I reverted it to the old way of ours - a simple deleteLater().

> Also, I am also concerned with QDialog::exec() being a blocking function.....

I don't really get this, we aren't starting the download until open/save is clicked (unless you mean the Download Books functionality, which I agree could've been an issue and already reverted...)